composer.json

```diff
@@ -31,20 +31,6 @@
         "symfony/process": "^3.0 || ^4.0",
         "symfony/stopwatch": "^3.0 || ^4.0"
     },
-    "require-dev": {
-        "johnkary/phpunit-speedtrap": "^1.1 || ^2.0 || ^3.0",
-        "justinrainbow/json-schema": "^5.0",
-        "keradus/cli-executor": "^1.2",
-        "mikey179/vfsstream": "^1.6",
-        "php-coveralls/php-coveralls": "^2.1",
-        "php-cs-fixer/accessible-object": "^1.0",
-        "php-cs-fixer/phpunit-constraint-isidenticalstring": "^1.1",
-        "php-cs-fixer/phpunit-constraint-xmlmatchesxsd": "^1.1",
-        "phpunit/phpunit": "^5.7.27 || ^6.5.14 || ^7.1",
```

Comment:
why "must"? what is the problem if they are not?

Reply:
that is a purpose of `require-dev`.
we are extending classes from phpunit package in our codebase - thus it must be there in composer.json file. if we use the class in src - then package should be in `dependencies`. if not, then in `devdependencies`.

removing dependencies directly used in code (dev or src) from composer file i consider a very bad practice. it is not the external tool that we are using out of code, like phpmd.

Reply:
maybe you should just suggest to Composer to drop the `require-dev` section at all?

Reply:
Purpose of `require-dev` is to lists packages required for developing this package - so following that reasoning all dependencies from `dev-tools` should be moved to main `composer.json`.

> if we use the class in src - then package should be in `dependencies`.

Is that rule about the src directory? ;) I believe (no-dev) dependencies are the ones needed in production - which in case of library is release - such as PHPUnit - because [some files](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.0/composer.json#L57) relying on it are in the release. Yet, no one ever consider moving it from `require-dev` to `require`.

> removing dependencies directly used in code (dev or src) from composer file i consider a very bad practice

I don't understand why? I mean the part for `dev`. For `src` it's obvious, but for `dev`? Even if PHP CS Fixer would be install via composer locally in project it's dev dependencies would be ignored.

TL;DR the idea was to move out dev dependencies from main `composer.json` to free normal dependencies from any potential conflict, nothing more.

Reply:
we don't care about conflicting dependencies of standalone tools, like phpmd. so we extracted it. cool.
we do care about conflicting dependencies of dev libraries we are using in our own codebase, eg phpunit, so let's keep them under main composer.json.